nucypher/crypto/kits.py

```diff
@@ -76,3 +80,78 @@ def __init__(self, *args, **kwargs) -> None:
     def from_bytes(cls, some_bytes):
         capsule, sender_pubkey_sig, ciphertext = cls.split_bytes(some_bytes)
         return cls(capsule=capsule, sender_pubkey_sig=sender_pubkey_sig, ciphertext=ciphertext)
+
+
+Revocation = namedtuple('Revocation', ['prefix', 'arrangement_id', 'signature'])
+
+
+class RevocationKit:
+
+    def __init__(self, policy_treasure_map):
+        self.revocations = dict()
+        for node_id, arrangement_id in policy_treasure_map:
+            self.revocations[node_id] = Revocation(b'REVOKE-', arrangement_id,
+                                                   constants.NOT_SIGNED)
+
+    def __iter__(self):
+        return iter(self.revocations.values())
+
+    def __getitem__(self, node_id):
+        return self.revocations[node_id]
+
+    def __len__(self):
+        return len(self.revocations)
+
+    def __eq__(self, other):
+        return self.revocations == other.revocations
+
+    @staticmethod
+    def revocation_to_bytes(revocation):
+        prefix, arrangement_id, signature = revocation
+        return b''.join((prefix, arrangement_id, bytes(signature)))
+
+    @staticmethod
+    def revocation_from_bytes(revocation_as_bytes: bytes):
+        from nucypher.crypto.signing import Signature
+        revocation_splitter = BytestringSplitter((bytes, 7),
+                                                 (bytes, 32),
+                                                 Signature)
+        return Revocation(*revocation_splitter(revocation_as_bytes))
+
+    @staticmethod
+    def verify_revocation(revocation, alice_pubkey: 'UmbralPublicKey'):
+        """
+        Verifies the revocation was from the provided pubkey.
+        """
+        from nucypher.crypto.signing import InvalidSignature
+
+        if not revocation.signature.verify(revocation.prefix + revocation.arrangement_id,
+                                           alice_pubkey):
+            raise InvalidSignature("Revocation has an invalid signature: {}".format(signature))
+        return True
+
+    @property
+    def revokable_addresses(self):
+        """
+        Returns a Set of revokable addresses in the checksum address formatting
+        """
+        return set(self.revocations.keys())
+
+    def sign_revocations(self, signer):
+        """
+        Signs all the revocations provided in the kit and adds them to
+        the revocations dict ready for use.
+        """
+        for node_id, revocation in self.revocations.items():
+            signature = signer(revocation.prefix + revocation.arrangement_id)
+            self.revocations[node_id] = Revocation(revocation.prefix,
+                                                   revocation.arrangement_id,
+                                                   signature)
+
+    def add_receipt(self, node_id, signed_receipt):
+        """
+        Adds a signed receipt of Ursula's ability to revoke the arrangement.
+        """
+        # TODO: Verify Ursula's signature
```

Comment:
Any details on this `TODO`?

Reply:
Sure, the overarching idea of a revocation receipt is that Ursula would sign the _signed revocation_ after performing the revocation. She would then return this signed receipt to Alice as output of the `DELETE` call. This is her agreement that she performed the revocation.

This `add_receipt` method gives Alice, or whoever performing the revocations, the ability to verify Ursula's signature as a means to verify that she agreed to remove the KFrag. What she does with this receipt is sorta an open question.

Reply:
...I think "receipt" was better.  What is being "confirmed" here?  All Ursula can really do is ack receipt of the rfra... notice.

Reply:
@jMyles - Were worried about nameshadowing here.   I suppose confirmation isnt much better.